src/Rings/ReesAlgebra.jl

```diff
@@ -0,0 +1,185 @@
+    var_names::Vector{String}=["s$i" for i in 0:ngens(M)-1]
+  )
+  R = base_ring(M)
+  r = rank(M)
+  S, s = PolynomialRing(R, Symbol.(var_names))
+  #S, _ = grade(S_tmp)
+  return S
+end
+
+function rees_algebra(M::SubQuo; 
+    var_names::Vector{String}=["s$i" for i in 0:ngens(M)-1],
+    check::Bool=true
+  )
+  success, p, sigma = is_projective(M)
+  if success
+    # The easy case: The Rees algebra is simply a polynomial ring 
+    # modulo linear equations in the variables parametrized by the base.
+    R = base_ring(M)
+    r = ngens(M)
+    S, s = PolynomialRing(R, Symbol.(var_names))
+    A = matrix(compose(sigma, p)) # The projector matrix:
+    # M is a direct summand of a free module via p : F ↔ M : sigma.
+    # Hence, the composition sigma ∘ p is the internal projection 
+    # of F onto M as a direct summand. 
+    #
+    # For a point x ∈ Spec(R) one has A(x) a matrix with entries in a field 𝕜
+    # Its span is the subspace M(x) ⊂ 𝕜ʳ, the fiber of M at x. 
+    # Polynomial functions on that fiber are polynomial functions 
+    # on 𝕜ⁿ in variables s₀,…,sᵣ₋₁, but restricted to M(x). Hence, we 
+    # can calculate modulo the ideal generated by linear forms 
+    # l = a₁s₁ + … + aᵣ₋₁sᵣ₋₁ vanishing on M(x). 
+    # 
+    # Since A(x) is a projector (A² = A), we find that the span M(x)
+    # is annihilated by the matrix B = 1 - A and that the columns of B 
+    # generate that ideal. 
+    B = one(A) - A
+    I = ideal(S, [sum(B[j, i]*s[j] for j in 1:length(s)) for i in 1:ncols(B)])
+    Q, sq = quo(S, I)
+    return Q
+  end
+
+  # The complicated case. We construct a versal morphism f : M → F to a 
+  # free module along the lines of Proposition 1.3 of [1] (in its published version).
+  return rees_algebra(_versal_morphism_to_free_module(M), check=false)
+end
+
+### Auxiliary methods needed for the Rees algebras
+
+### The following function is implemented along the lines of Proposition 1.3 of [1] in 
+# its published version.
+function _versal_morphism_to_free_module(M::SubQuo)
+  R = base_ring(M)
+  R1 = FreeMod(R, 1)
+  M_double_dual, psi = double_dual(M, cod=R1)
+  M_dual = domain(element_to_homomorphism(zero(M_double_dual)))
+  pres_M_dual = presentation(M_dual)::ChainComplex
+  g = map(pres_M_dual, 0) # The projection of a free module onto M_dual
+  g_dual = dual(g, cod=R1, codomain_dual=M_double_dual)
+  return compose(psi, g_dual)
+end
+
+function proj(S::MPolyRing_dec)
+  is_standard_graded(S) || error("ring must be standard graded")
+  return ProjectiveScheme(S)
+end
+
+function proj(Q::MPolyQuo{<:MPolyElem_dec})
+  S = base_ring(Q)
+  is_standard_graded(S) || error("ring must be standard graded")
+  return ProjectiveScheme(Q)
+end
+
+#function is_injective(f::ModuleFPHom) 
+#  return iszero(kernel(f)[1])
+#end
+#
+#function is_surjective(f::ModuleFPHom)
+#  return iszero(cokernel(f))
+#end
+
+function is_isomorphism(f::ModuleFPHom)
+  return is_injective(f) && is_surjective(f)
+end
+
```

Comment:
Do the '#' in front of is_injective and is_surjective mean that this functionality is inherited from somewhere else? They are used in is_isomorphism after all.

Reply:
Yes, it turned out that they were already there. Partially disfunctional, though, as it seems. See the tests in part I.

Reply:
So 129-135 could be removed before merge, right?